fix(person): avoid repeated last name via new last_name_patterns

matthewmayer opened the pull request:
fix #1817

an alternative strategy which allows for compound and hyphenated last names. The advantage of this is that it works for both `faker.person.lastName()` as well as `faker.person.fullName()`

```js
faker.locale="es"

faker.person.lastName() // 'Garibay Piña'
faker.person.fullName() // 'Marta Menchaca Vergara'
```



<details>
<summary>Here's 100 fullNames for each of the affected locales</summary>

```js
["es","es_MX","en_GH", "lv", "nb_NO", "sv"].forEach(locale => {faker.locale = locale; [...Array(100)].map(x=>console.log(locale, faker.person.fullName()))})
```
```
es Pedro Monroy Navarro
es Sra. Roser Vaca Rendón
es Lorena Valencia Marrero
es Pedro Arteaga Nevárez
es Pablo Duarte Pelayo
es Sra. Maica Cotto Feliciano
es Lorena Urrutia Velasco
es Pedro Juárez Casares
es Sra. Anni Abreu Lucero
es Iván Zelaya Almaráz
es Pablo Palacios Lucio
es Andrea Henríquez Verduzco
es Miguel Olmos Cano
es Iván Pedroza Olivera
es Lorena Estévez Zayas
es Salvador Sierra Mojica
es Roser Muro Navarro
es Ricardo Anaya Medina
es Ángel Abreu Balderas
es María de los Ángeles Salas Rivero
es Anni Carrillo Lemus
es Jennifer Villaseñor Tello
es Miguel Altamirano Laboy
es Laura Cortéz Rolón
es Jordi Gurule Nava
es Sra. Maica Leyva Colunga
es Anni Menchaca Colón
es Maica Lugo Alonso
es Andrea Prieto Lovato
es Jorge Maldonado Medina
es Ricardo Montalvo Armas
es Carles Zepeda Linares
es María de los Ángeles Velasco Quintanilla
es Anni Corona Casares
es Pío Meraz Olivas
es Lorena Gurule Atencio
es Lorena Aguilera Carrero
es Hermenegildo Rosado Alanis
es María de los Ángeles Mercado Negrón
es Jennifer Deleón Muñoz
es Salvador Brito Santana
es Ana Puga Villaseñor
es Jorge Nevárez Serna
es Lorena Garica Lomeli
es Sra. Andrea Quiñónez Osorio
es Pablo Torres Reséndez
es Jennifer Huerta Niño
es Josep Chávez Tapia
es Anni Matías Aparicio
es Matilde Escalante Vega
es Juan Ramón Peña Luevano
es Daniel Araña Jáquez
es Matilde Bonilla Viera
es Jennifer Cuellar Zelaya
es Carles Espinoza Sauceda
es Roser Soliz Brito
es Andrea Sánchez Escalante
es Lorena Delgado Valladares
es Sergi Soto Ybarra
es Sergi Alvarado Cadena
es Iván Meléndez Armenta
es Pío Flores Alaníz
es Josep Zamudio Pelayo
es Josep Guardado Quesada
es Jennifer Bravo Malave
es Ángel Crespo Espinal
es Sergi Zelaya Calvillo
es Jennifer Alvarado Berríos
es Jordi Santacruz Barrios
es Roser Hernández Amaya
es Pío Urrutia Gil
es Ángel Barreto Tovar
es Matilde Correa Collazo
es Daniel Tamez León
es Manuel Tovar Rosario
es Pedro Badillo Haro
es Pedro Brito Márquez
es Jordi Holguín Salazar
es Jennifer Quezada Vega
es Marta Saldaña Estrada
es Laura Ponce de Jesús
es Daniel Delgado Esquivel
es María de los Ángeles Escalante Amaya
es Carles Meza Raya
es Lorena Meléndez Laureano
es Iván Alcántar Escamilla
es Maica Gurule Amaya
es Miguel Zarate Batista
es Sr. Jorge Escobedo Alonzo
es Ricardo Puga Lozano
es Salvador Bahena Enríquez
es Ricardo Girón Huerta
es Matilde Benítez Lara
es Ana Meza Rico
es Miguel Molina Delapaz
es Anni Báez Jiménez
es Sta. Anni Cortés Gálvez
es Manuel Acuña Flórez
es Salvador Hidalgo Corrales
es Juan Ramón Vela Solorzano
es_MX Hugo Urías Villarreal
es_MX Florencia Santillán Barela DVM
es_MX Caridad Delgadillo Riojas
es_MX Carmen Casares Calderón
es_MX Miguel Ángel Partida Raya
es_MX Victor Manuel Rojo de Deleón
es_MX Julio Gollum Peña
es_MX Eva Frías Aponte
es_MX Margarita Espinal Trujillo
es_MX Cristian Contreras Peña
es_MX Ivanna Cardona Contreras
es_MX Julia Mata de Soto
es_MX Gabriela Verduzco Curiel
es_MX Francisco Vanegas de Regalado
es_MX Lorena Barajas Jaimes
es_MX Leticia Calvillo Cerda
es_MX Axel Terán Córdova
es_MX Antonio Solano López
es_MX Jennifer Salinas Zamarreno
es_MX Xochitl Hernández Borrego
es_MX Mónica Guzmán Vargas
es_MX Guillermina Vanegas Centeno
es_MX Ariadna Villaseñor Rodríguez
es_MX Olivia Meza Suárez
es_MX Mariano Alcala Tamayo
es_MX Sara Samaniego Fonseca
es_MX Roberto Covarrubias de Soliz
es_MX Evelyn Enríquez Carrero
es_MX Carlos Escalante Avilés
es_MX Sta. Amalia Domínguez de Salinas
es_MX Sr. Juan Manuel Ozuna Duarte
es_MX Esperanza Alanis Sauceda
es_MX Horacio Palomo Orta
es_MX Alfonso Barela Munguía
es_MX Leticia Barragán Muñiz
es_MX Gael Beltrán Alva Jr.
es_MX Yolanda Xiana Gastélum
es_MX Abigail Godoy Mota
es_MX Mariana Saavedra de Arenas
es_MX Cecilia Delapaz Yunta Dr.
es_MX Laura Alcaraz de Lozada
es_MX Óscar Ojeda de Matías
es_MX Florencia Ponce Caraballo
es_MX Alexis Caballero Gaona
es_MX Sancho Quintana Santillán
es_MX Víctor Benítez Valdez Sr.
es_MX Micaela Alvarez Bustos
es_MX César Borrego Mendoza
es_MX Uriel Mares de Bustos
es_MX Vicente Maldonado de Valdez IV
es_MX Martín Zepeda Véliz
es_MX Andrés García Salinas
es_MX Antonia Tijerina Granado
es_MX María Cristina Gollum Kanaria
es_MX Juan Manuel Castellanos de Huixtlacatl
es_MX Rocio Enríquez Valles
es_MX Gloria Santillán Dávila Ing.
es_MX Sr. Luis Fernando Hurtado Rojo
es_MX Emmanuel Dueñas Garrido
es_MX Leonor Solorzano de Armenta
es_MX José Emilio Figueroa Mora Mtro.
es_MX María Fernanda Mejía Cavazos PhD
es_MX Elizabeth Lemus Villalpando Ing.
es_MX Paulina Delrío Caldera
es_MX Claudio Casárez Brito
es_MX Marisol Ruelas Kindelan
es_MX Ángel Daniel de Anda Estrada
es_MX Juan Manuel Briseño Anguiano
es_MX Jimena Quiros de Rosales
es_MX Dorotea Guerra Manzanares
es_MX Mayte Pichardo Baeza Sr.
es_MX Sr. Raúl Palacios Mora
es_MX Axel Bahena Villagómez
es_MX José Luis Ojeda Lebrón Lic.
es_MX Mauricio Lemus Navarro
es_MX Antonia Arellano de Carrasco
es_MX Clemente Abreu Peralta Dr.
es_MX Fernando Javier Montez Karan
es_MX Victor Manuel Grijalva de Toro
es_MX Luis Miguel Negrón de Solorio
es_MX Rosa Delao Granado Sr.
es_MX Tadeo Riojas Tafoya V
es_MX Marco Antonio Guevara Cepeda
es_MX Francisco Corral Olivas
es_MX Lucia Espino Sáenz
es_MX Samuel Holguín de Borrego Mtro.
es_MX Sr. Jesús Roque Vaca
es_MX Cristobal Viera Macías
es_MX Sra. Reina Gurule Becerra
es_MX Carla Altamirano Linares
es_MX Iker Varela de Vaca
es_MX Susana Ceja Huerta
es_MX Jacobo Cintrón de Manzanares
es_MX Esperanza Quezada Delgado
es_MX Luz Rico de Posada
es_MX María Teresa Canales Valverde
es_MX Juan Pablo Duran Cornejo
es_MX Ariadna Acevedo Lozada
es_MX Beatriz Mercado Robledo IV
es_MX Abril Portillo Marrero
en_GH Millicent Boakye
en_GH Adrian Opuku
en_GH Kenneth Adu
en_GH Russell Kumi-Kyei
en_GH Promise Adadevoh
en_GH Awesi Afoakwa
en_GH Henry Nti-Ntiamoa
en_GH John Agyare-Ankra
en_GH Donald Andorful
en_GH Derrick Yawson-Afirifa
en_GH Naomi Nyantakyi
en_GH Helen Adutwum-Tutu
en_GH Eunice Yirenkyi
en_GH Howard Agyapong-Karikari
en_GH Timothy Okpattah-Boahen
en_GH Eric Awuah
en_GH Douglas Adongo
en_GH Thy Will Be Done Quartey-Oppong
en_GH Abigail Boateng-Badu
en_GH Steven Amoa
en_GH Abigail Daako-Djokoto
en_GH Awesi Karikari
en_GH Kenneth Amo-Kusi
en_GH Oliver Opoku-Nkansa
en_GH Maria Ofori-Yeboa
en_GH Promise Duah
en_GH Dorothy Bona-Yawson
en_GH Katherine Nyaako-Ansa
en_GH Jessica Nkansa
en_GH Ekow Bonsra-Akyeampong
en_GH Charlotte Bonsra-Adongo
en_GH Derrick Nkansah
en_GH Kwamena Adu
en_GH Jason Safo-Asenso
en_GH Kate Koomson
en_GH Jeffrey Ofori
en_GH Jeffrey Nyantakyi
en_GH Abena Tawia-Dwamena
en_GH Mandy Akyena
en_GH Francesca Asare-Amoa
en_GH Rebecca Nyantakyi
en_GH Catherine Ampadu-Okyere
en_GH Karen Mensa-Sarpong
en_GH Lydia Ofori
en_GH Kenneth Akyena-Asamoah
en_GH Esi Adongo-Asiama
en_GH James Dzokoto-Tutu
en_GH Eyram Asare
en_GH Joojo Kwarteng
en_GH Maureen Wiafe-Tawia
en_GH Amber Baafi
en_GH Howard Acheampong-Adadevoh
en_GH Nimakoah Koomson-Amoasi
en_GH Georgina Obeng-Oti
en_GH Henry Osei
en_GH Lisa Adadevoh
en_GH Wendy Nyaako-Agyei
en_GH Lisa Nkansa-Otuo
en_GH Regina Kwarteng
en_GH Gemma Mensah-Fofie
en_GH Jacob Tweneboa
en_GH Kweku Bona
en_GH Kevin Safo
en_GH Nigel Asenso
en_GH Susan Amoako
en_GH Kwame Amoa
en_GH Veronica Danso-Antwi
en_GH Katherine Yeboah
en_GH Emma Asiama-Boakye
en_GH Jessica Dzokoto-Adongo
en_GH Sandra Bona
en_GH Mark Agyare
en_GH Max Sarpong
en_GH Julian Nkrumah
en_GH Ama Opoku
en_GH Promise Mensah-Daako
en_GH Kenneth Fosu-Kumi
en_GH Yvonne Kwaakye
en_GH Akumaa Daako
en_GH Joe Omani
en_GH Akofa Appiah-Kwarteng
en_GH Bless Tweneboa
en_GH Sheila Nyantakyi
en_GH Lisa Adongo-Duah
en_GH Frank Yeboah
en_GH Millicent Akowua-Afirifa
en_GH Nicola Bona-Safo
en_GH Thank God Kumi-Omani
en_GH Naomi Gyamfi-Ata
en_GH Marian Boakye-Nyame
en_GH Joshua Ankra-Owusu
en_GH Timothy Tweneboa-Dzokoto
en_GH Adam Akoto-Adomako
en_GH Victoria Afirifa
en_GH Henry Adomah-Appia
en_GH Martin Otuo
en_GH Awesi Mensa
en_GH Yvonne Darko-Gyasi
en_GH Grace Boateng-Safo
en_GH Clifford Nsia
lv Jūlijs Krūmiņš
lv Aurēlija Sprūde
lv Konrāds Pavlovskis
lv Jurģis Dzirkalis
lv Beāte Birzniece-Danielsone
lv Trīne Sietniece-Liepiņa
lv Jāzeps Danielsons-Blekte
lv Tamāra Porgante
lv Dr. Rūdolfs Viļums-Aigars
lv Valentīna Loča-Alksne k-dze
lv Ojārs Matisons
lv Ārija Liepa-Krieviņa
lv Kristīne Krieva
lv Boļeslavs Ostrovskis
lv Evelīna Eglīte kundze
lv Ermīns Zībers
lv Niklāvs Beļavskis
lv Mārtiņš Aigars
lv Ēvalds Runcis
lv Biedrs Justīne Zībere
lv Mārcis Gibala
lv Janīna Miķelsone-Sondore
lv Klāvs Tučs kundze
lv Biedrs Zigrīda Ķeizare
lv Renārs Sīlis
lv Luīze Birziņa-Loča
lv Aīda Zariņa
lv Vijolīte Tuča
lv Andžejs Putniņš
lv Ziedīte Lielmane kundze
lv Ilmārs Vītiņš
lv Janīna Zvinele
lv Liāna Kārkliņa
lv Modrīte Čudara
lv Emīlija Vītola
lv Andris Eglītis
lv Beāte Vītoliņa kundze
lv Uģis Miķelsons
lv Eduards Meškūns
lv Biedrs Zigfrīda Lāce
lv Jūlijs Tiltiņš
lv Dr. Dzīle Blaua
lv Biedrs Māris Dreimanis
lv Ēvalds Siliņš
lv Emīlija Rozenbaha
lv Indriķis Smuģis
lv Romāns Balodis-Ivanovs
lv Ernestīne Zviedre
lv Žanna Cīrule
lv Klāra Vanaga
lv Jasmīna Meiere kundze
lv Eižens Priedītis
lv Paulīne Legzdiņa k-dze
lv Valērijs Kauliņš-Salmiņš
lv Dr. Gunārs Podziņš
lv Dzirkstīte Krieviņa
lv Prof. Hermīne Jumiķe
lv Lība Zālīte k-dze
lv Biedrs Einārs Putniņš
lv Pāvils Zunda-Meiers
lv Pāvils Bušs
lv Eduards Lielmanis
lv Modrīte Tuča
lv Jēkabs Kauliņš
lv Modrīte Porgante
lv Biedrs Pārsla Zariņa
lv Ermīns Vītols
lv Edīte Zviedre
lv Biedrs Mārīte Dreimane
lv Dr. Ervīns Celmiņš
lv Prof. Kristiāna Grūba
lv Ermīns Rožkalns
lv Pāvils Pavlovskis
lv Jāzeps Meiers
lv Rudīte Arāja
lv Regīna Miķelsone
lv Ādams Valters-Bīriņš
lv Jūsma Ostrovska-Liepiņa
lv Prof. Zīle Zeidmane
lv Dzirkstīte Veisa
lv Einārs Teicāns
lv Lavīze Auziņa
lv Mārtiņš Kulmanis k-dze
lv Dāgs Ķēniņš
lv Prof. Edžus Latkovskis
lv Pēteris Kļaviņš
lv Armīns Zālītis
lv Ādams Namiķis
lv Līze Veinberga-Muceniece
lv Inārs Kronbergs
lv Andžs Celmiņš
lv Līvija Gasjūne
lv Dzīle Kārkliņa kundze
lv Akvelīna Kalniņa kundze
lv Rūdolfs Krauze
lv Kristiāns Alksnis-Lodiņš
lv Jurģis Priedītis
lv Helēna Ostrovska
lv Leontīne Lodiņa-Ozere
lv Helēna Briede-Muižniece kundze
nb_NO Dr. Mina Henriksen
nb_NO Kristoffer Jørgensen
nb_NO Synne Dahl Nilsen
nb_NO Sebastian Skoglund
nb_NO Kristoffer Lunde Ruud
nb_NO Isak Arnesen
nb_NO Elias Danielsen
nb_NO Mari Krogh Gundersen
nb_NO Anders Huseby
nb_NO Celine Huseby
nb_NO Dr. Andreas Johannessen
nb_NO Aurora Johannessen Østby
nb_NO Martin Fjeld Aas
nb_NO Johannes Jakobsen
nb_NO Jenny Fredriksen
nb_NO Elise Nguyen
nb_NO Helene Røed
nb_NO Jenny Nielsen IV
nb_NO Tobias Borge
nb_NO Maria Johnsen Hansen
nb_NO Sunniva Carlsen
nb_NO Marius Karlsen
nb_NO Mats Tangen
nb_NO Oda Martinsen
nb_NO Benjamin Nordskaug Stensrud Jr.
nb_NO Lucas Nordskaug
nb_NO Camilla Pedersen Karlsen
nb_NO Aurora Svendsen
nb_NO Jørgen Nielsen
nb_NO Simen Kvarme
nb_NO Dr. Filip Skoglund
nb_NO Kaja Solheim
nb_NO Lucas Tangen
nb_NO Kristoffer Berge Olstad
nb_NO Markus Aas
nb_NO Maren Myhrer Jr.
nb_NO Mari Strand
nb_NO Prof. Mia Borge
nb_NO Kasper Berntsen
nb_NO Benjamin Andreassen
nb_NO Ole Gran I
nb_NO Oskar Pedersen
nb_NO Theodor Martinsen Haugen
nb_NO Selma Jensen Østby Jr.
nb_NO Kristine Martinsen
nb_NO Martin Andreassen
nb_NO Joakim Nygård Jørgensen
nb_NO Kristine Hagen
nb_NO Mikkel Huseby
nb_NO Victoria Jacobsen
nb_NO Lars Svendsen
nb_NO Dr. Linnea Lie
nb_NO Linnea Eriksen
nb_NO Martine Bjerke
nb_NO Synne Nygård
nb_NO Oskar Holm
nb_NO Eline Ryan
nb_NO Julie Jensen Jakobsen
nb_NO Tuva Ruud
nb_NO Jenny Pedersen
nb_NO Ida Svendsen
nb_NO Nikolai Bjerke
nb_NO Leah Nordby Bakken
nb_NO Jonas Torgersen
nb_NO Jenny Finstad
nb_NO Marte Østby
nb_NO Victoria Dahl
nb_NO Maren Andersen
nb_NO Karoline Røed Moe
nb_NO Marie Kristiansen
nb_NO Silje Nordskaug
nb_NO Mia Haugen
nb_NO Oliver Nguyen Hagen
nb_NO Noah Løken
nb_NO Sara Sørlie
nb_NO Dr. Marte Johansen
nb_NO Simen Fossum
nb_NO Fredrik Skoglund II
nb_NO Kristine Jensen
nb_NO Sander Aas
nb_NO Kristian Bjerke Nordby
nb_NO Sebastian Østli Eriksen
nb_NO Natalie Andreassen III
nb_NO William Krogh
nb_NO Linnea Fjeld IV
nb_NO Fredrik Glosli
nb_NO Dr. Kristine Ødegård
nb_NO Mikkel Jacobsen
nb_NO Joakim Kristensen V
nb_NO Dr. Andreas Strand Fredriksen
nb_NO Amalie Johannessen
nb_NO Mari Martinsen
nb_NO Madeleine Danielsen
nb_NO Ida Moen
nb_NO Emil Johnsen
nb_NO Lucas Nguyen
nb_NO Prof. Helene Tangen
nb_NO Prof. Mia Sæther
nb_NO Alexander Vedvik
nb_NO Celine Berg
sv Simon Sjöberg Berggren
sv Elin Eriksson
sv Dennis Ekström
sv Magnus Sandberg
sv Amanda Samuelsson
sv Moa Olofsson
sv Linn Hermansson
sv Prof. Charlotte Falk
sv Inger Mohamed
sv Britt Sjögren II
sv Andreas Dahlberg Lund
sv Maja Abrahamsson MD
sv Elin Norberg
sv Maria Dahl
sv Henrik Persson
sv Ali Holmberg
sv Axel Björklund
sv Joel Wikström Bengtsson
sv Kurt Lund
sv Karl Karlsson
sv Liam Nordin
sv Pia Hellström
sv Bengt Gunnarsson
sv Erika Lundin
sv Christoffer Sandström
sv Madeleine Nordin Bengtsson DDS
sv Joakim Magnusson
sv Samuel Persson
sv Erik Olsson Hassan
sv Jakob Johansson DVM
sv Pernilla Pålsson
sv Britt-Marie Lundgren
sv John Blom
sv Sven Nordström IV
sv Nina Sandström
sv Camilla Nyström Lindqvist
sv Amanda Jonsson Hedlund
sv Julia Ali Viklund
sv Oliver Lundberg
sv Barbro Åberg
sv Joel Berggren
sv Marianne Olofsson
sv Mikael Åkesson
sv Martin Persson IV
sv Sofia Olofsson
sv Mikael Henriksson
sv Anneli Arvidsson Jansson
sv Astrid Hedlund
sv Agneta Magnusson
sv Åke Wikström
sv Nils Öberg Karlsson
sv Rickard Ström
sv Prof. Edvin Lindqvist
sv Lena Björk
sv Johnny Lind
sv PhD. Elisabeth Arvidsson
sv Hugo Nyberg I
sv Lars Mattsson Holmgren
sv Gunnel Abrahamsson Forsberg
sv Fredrik Håkansson
sv Sofie Henriksson
sv Jonas Karlsson
sv Rolf Lindqvist
sv Josef Nilsson Hansson
sv Elsa Lund
sv Charlotte Lundström
sv Cecilia Forsberg
sv Simon Dahl
sv Julia Mohamed
sv Sven Bergqvist DVM
sv Gunnar Gustafsson
sv Josefin Jakobsson PhD
sv Lars Lundström
sv Monica Holm
sv Johnny Isaksson
sv Roger Falk
sv Filip Sandström
sv Hanna Karlsson Lundgren
sv Louise Johansson
sv Lina Lindgren
sv Felix Sundberg Jr.
sv Ludvig Löfgren
sv Mikaela Isaksson
sv Helen Hedlund
sv Ann-Christin Öberg
sv Jakob Hassan
sv Dr. Lucas Sundberg
sv Felix Isaksson
sv Ulla Holmberg
sv Prof. Robin Holmberg
sv Gunnel Åkesson V
sv Linus Mårtensson
sv Ellen Söderberg Blomqvist
sv Samuel Dahl
sv Malin Sjögren
sv Agneta Lindholm
sv Alva Lundberg
sv Stig Lindqvist Håkansson
sv Britt Dahlberg
sv Carina Nordin
```
</details>

matthewmayer committed:
fix(person): avoid repeated double last name via new last_name_patterns

matthewmayer commented on src/locales/nb_NO/person/name.ts:
```diff
@@ -7,13 +7,5 @@ export default [
     value: '{{person.firstName}} {{person.lastName}} {{person.suffix}}',
     weight: 1,
   },
-  {
-    value: '{{person.firstName}} {{person.firstName}} {{person.lastName}}',
-    weight: 1,
-  },
```
removing double firstNames for now. This should be addressed by middleName if appropriate, but nb_NO has no middle names.

matthewmayer commented on src/locales/lv/person/name.ts:
```diff
@@ -8,13 +8,4 @@ export default [
     weight: 1,
   },
   { value: '{{person.firstName}} {{person.lastName}}', weight: 7 },
-  { value: '{{person.lastName}} {{person.firstName}}', weight: 2 },
```
this seems wrong, i couldnt find any evidence Latvian names are written reversed

matthewmayer commented on src/locales/lv/person/name.ts:
```diff
@@ -8,13 +8,4 @@ export default [
   },
   { value: '{{person.firstName}} {{person.lastName}}', weight: 7 },
-  { value: '{{person.lastName}} {{person.firstName}}', weight: 2 },
-  {
-    value: '{{person.firstName}} {{person.firstName}} {{person.lastName}}',
-    weight: 2,
-  },
```
again, remove double first names

matthewmayer commented on src/modules/person/index.ts:
```diff
@@ -111,11 +111,22 @@ export class PersonModule {
-      female: female_last_name,
-      male: male_last_name,
-    });
+    const makeLastName = () =>
+      selectDefinition(this.faker, sex, {
+        generic: last_name,
+        female: female_last_name,
+        male: male_last_name,
+      });
```
there might be a neater way to do this

matthewmayer commented on src/locales/es_MX/person/name.ts:
```diff
@@ -1,20 +1,14 @@
+    weight: 8,
   },
   {
-    value:
-      '{{person.suffix}} {{person.firstName}} {{person.lastName}} {{person.lastName}}',
```
also seems like an error, suffix shouldnt be at the start!

ST-DDT commented on src/locales/en_GH/person/last_name_patterns.ts:
```diff
@@ -0,0 +1,4 @@
+export default [
+  { value: '{{person.lastName}}', weight: 1 },
+  { value: '{{person.lastName}}-{{person.lastName}}', weight: 1 },
```
IMO these should refer directly to `person.last_name` and be used with fake, so it is possible to use custom patterns here as well just as the Dutch tusselvogsel.

But for that we would need #1677 first.

matthewmayer replied:
I don't have a particular preference for using `fake` vs `mustache` but i dont think this mechanism should be used for tussenvoegsels, per the comment of @amarillion on  https://github.com/faker-js/faker/issues/1777

> The way faker used to do this, wasn't really realistic. Affixes are part of the family name. The way the old version of faker would merge a random affixless family name with a random affix isn't really realistic. "van der Meer" (lit: from the lake) is a valid Dutch name, "Kuipers" (cooper) as well. "van der Kuipers" (from the cooper???) is nonsensical.

ST-DDT replied:
Thats true for the Dutch names, but the last names might be split by/a combination of "female" and "male" lastname or "von {{location.city_name}}".

matthewmayer replied:
hmm but if you change the method to something like this to use fake, how do you make it gender-aware?

```js
lastName(sex?: SexType): string {
    if (this.faker.definitions.person.last_name_patterns) {
      const pattern = this.faker.helpers.weightedArrayElement(
        this.faker.definitions.person.last_name_patterns
      );
      return this.faker.helpers.fake(pattern); //how do you make this gender aware?
    } else {
      const { last_name, female_last_name, male_last_name } =
        this.faker.definitions.person;
      return selectDefinition(this.faker, sex, {
        generic: last_name,
        female: female_last_name,
        male: male_last_name,
      });
    }
  }
```

ST-DDT replied:
You split between genders on the pattern level.

Person.last_name_pattern.female/male/generic.
Thats why I mentioned #1677.
Sorry for my unclear communication.

matthewmayer replied:
Currently the locales with hyphenated/compound surnames are {en_GH, es, es_MX, lv, nb_NO, sv} and the locales with gendered surnames are {sk, id_ID, lv, cz, ru, uk, dv, mk, az} so i think the only current overlap is lv.

So for example if a Latvian has a hyphenated female-gendered surname, are both parts female?

matthewmayer replied:
I think we could do it this way for now and leave a TODO to revisit after #1677

ST-DDT commented:
**Team Decision**

We will use this approach, but we would like to use the fake function (in combination with person.last_name) instead of mustache, to allow custom pattern parts.

We will revisit this/the pattern as part of #1677 .

matthewmayer commented:
So this is temporarily on hold until #1677 is implemented?

ST-DDT commented:
> So this is temporarily on hold until #1677 is implemented?

Sorry, for not being precise enough with my comment.
No, you can implement it now with:

````ts
female_last_name_pattern, male_last_name_pattern, last_name_pattern
````

With #1677 we will change it to:

````ts
last_name_pattern: {female, male, generic}
````

matthewmayer committed:
switch to use faker.helpers.fake

matthewmayer commented on src/modules/person/index.ts:
```diff
@@ -55,6 +55,72 @@ function selectDefinition(
+ * @param param2.male Male definitions.
+ * @returns Definition based on given sex.
+ */
+function selectWeightedDefinition(
```
I'm sure someone can suggest me a better way to handle this, this is needed because selectDefinition doesn't support weighted array definitions.

ST-DDT replied:
How about joining these two methods into one?

````ts
/**
 * Select a definition based on given sex.
 *
 * @param faker Faker instance.
 * @param elementSelectorFn The method used to select the actual element.
 * @param sex Sex.
 * @param param2 Definitions.
 * @param param2.generic Non-sex definitions.
 * @param param2.female Female definitions.
 * @param param2.male Male definitions.
 * @returns Definition based on given sex.
 */
function selectDefinition<T>(
  faker: Faker,
  elementSelectorFn: (values: T[]) => string,
  sex: SexType | undefined,
  // TODO @Shinigami92 2022-03-21: Remove fallback empty object when `strict: true`
  { generic, female, male }: { generic?: T[]; female?: T[]; male?: T[] } = {}
): string {
  let values: T[] | undefined;

  switch (sex) {
    case Sex.Female:
      values = female;
      break;

    case Sex.Male:
      values = male;
      break;

    default:
      values = generic;
      break;
  }

  if (values == null) {
    if (female != null && male != null) {
      values = faker.helpers.arrayElement([female, male]);
    } else {
      values = generic;
    }
  }

  return elementSelectorFn(values);
}
````

Then it can be used like this:

````ts
if (this.faker.definitions.person.last_name_patterns) {
  const pattern = selectDefinition(
    this.faker,
    this.faker.helpers.weightedArrayElement,
    sex,
    {
      generic: last_name_patterns,
      female: female_last_name_patterns,
      male: male_last_name_patterns,
    }
  );
  return this.faker.helpers.fake(pattern);
} else {
  return selectDefinition(
    this.faker,
    this.faker.helpers.arrayElement,
    sex,
    {
      generic: last_name,
      female: female_last_name,
      male: male_last_name,
    }
  );
}
````

ST-DDT commented on src/modules/person/index.ts:
```diff
@@ -108,14 +174,29 @@ export class PersonModule {
+      female_last_name_patterns,
+    } = this.faker.definitions.person;
+
+    if (this.faker.definitions.person.last_name_patterns) {
```
```suggestion
    if (last_name_patterns != null || male_last_name_patterns != null  || female_last_name_patterns != null ) {
```

Commit:
Update src/modules/person/index.ts

Co-authored-by: ST-DDT <ST-DDT@gmx.de>

Commit:
Merge branch 'next' into fix/person-last-name-patterns

matthewmayer committed:
make selectDefinition function generic

matthewmayer committed:
add fe/male_last_name_patterns to person definitions

matthewmayer committed:
also add hyphenated surnames for en_GB

about 10% of names are hyphenated in UK https://www.theguardian.com/lifeandstyle/2017/nov/02/keeping-up-with-smith-joneses-no-longer-posh-double-barrelled-surname

matthewmayer committed:
lint

ST-DDT committed:
Merge branch 'next' into fix/person-last-name-patterns

ST-DDT committed:
Merge branch 'next' into fix/person-last-name-patterns
